slm_porosity/heat_transfer.f90

```diff
@@ -335,58 +344,49 @@ FUNCTION user_rhs (u_integrated, p)
     IF (IMEXswitch .GE. 0) THEN
        CALL c_diff_fast(enthalpy_fusion(u_integrated(:,ie)), du, du_dummy, j_lev, ng, meth, 10, ne, 1, ne)
        T = temperature(u_integrated(:,ie))
-       du(ie,:,:) = du(ie,:,:) * SPREAD(conductivity(T) / capacity(T) * porosity_term(), 2, dim)
+       liquid_fraction = liquid_fraction(u_integrated(:,ie))
+       du(ie,:,:) = du(ie,:,:) * SPREAD(conductivity(T, liquid_fraction) / capacity(T, liquid_fraction) * porosity_term(), 2, dim)
        CALL c_diff_fast(du(ie,:,:), d2u, d2u_dummy, j_lev, ng, meth, 10, dim, 1, dim)
        DO i = 1, dim
           user_rhs(shift+1:shift+ng) = user_rhs(shift+1:shift+ng) + d2u(i,:,i)
        END DO
     END IF
-    !PRINT *, 'user_rhs', user_rhs
-
   END FUNCTION user_rhs
 
+  !\frac{d}{dh_*} = \frac{d}{d\phi} \frac{d\phi}{dh_*} = \frac{d}{d \phi} 1/\frac{dh_*}{d\phi} = \frac{d}{d \phi} 1/(h_fus+#!#!#d\phi/dh)
   FUNCTION user_Drhs (pert_u, u_prev, meth)
     IMPLICIT NONE
     INTEGER, INTENT(IN) :: meth
     REAL (pr), DIMENSION (ng,ne) :: pert_u, u_prev
     REAL (pr), DIMENSION (n) :: user_Drhs
-    REAL (pr), DIMENSION (ng) :: T
+    REAL (pr), DIMENSION (ng) :: T, liquid_fraction
```

Comment:
Now, I suggest to use `T, phi, c_p, k_0`. Short names are acceptable for local variables.